server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferMetadata.java

```diff
@@ -57,6 +65,16 @@ public TranslogTransferMetadata(long primaryTerm, long generation, long minTrans
         this.count = count;
     }
 
+    public TranslogTransferMetadata(StreamInput in) throws IOException {
+        InputStreamDataInput inputStreamDataInput = new InputStreamDataInput(in);
+        CodecUtil.checkHeader(inputStreamDataInput, METADATA_CODEC, CURRENT_VERSION, CURRENT_VERSION);
```

Comment:
This might not be enough, for corruption we should check for
`CodecUtil.checksumEntireFile(indexInput)`

Reply:
As of now, this is not done to check for corruption. We are reading and checking header so that stream can progress and we can read actual fields of the metadata.

Reply:
`CodecUtil.checksumEntireFile` takes IndexInput as an argument which requires reading all the bytes from InputStream before creating the IndexInput instance. Even though metadata file will not be huge in size, I am not sure if we should read the file fully for checksum.

Change is simple though. In `TranslogTransferManager`, instead of following code:

```
try (InputStreamStreamInput streamInput = new InputStreamStreamInput(transferService.downloadBlob(remoteMetadaTransferPath, filename))) {
    return new TranslogTransferMetadata(streamInput);
}
```

We need following:

```
try (InputStream inputStream = transferService.downloadBlob(remoteMetadaTransferPath, filename);) {
    IndexInput indexInput = new ByteArrayIndexInput("metadata file", inputStream.readAllBytes());
    return new TranslogTransferMetadata(indexInput);
}
```

Reply:
Metadata file should be of bounded size so I guess we should do a full checksum

Reply:
Got it, pushing the change in next commit.